src/main/user-impl/java/com/mysql/cj/jdbc/ha/plugins/Monitor.java

```diff
@@ -62,23 +60,22 @@ static class ConnectionStatus {
     }
   }
 
-  static final long DEFAULT_CONNECTION_CHECK_INTERVAL_MILLIS = 100;
-  static final long DEFAULT_CONNECTION_CHECK_TIMEOUT_MILLIS = 3000;
   private static final int THREAD_SLEEP_WHEN_INACTIVE_MILLIS = 100;
+  private static final int MIN_CONNECTION_CHECK_TIMEOUT_MILLIS = 3000;
   private static final String MONITORING_PROPERTY_PREFIX = "monitoring-";
 
-  private final Queue<MonitorConnectionContext> contexts = new ConcurrentLinkedQueue<>();
+  private final Queue<MonitorConnectionContext> activeContexts = new ConcurrentLinkedQueue<>();
+  private final Queue<MonitorConnectionContext> newContexts = new ConcurrentLinkedQueue<>();
   private final IConnectionProvider connectionProvider;
   private final Log logger;
   private final PropertySet propertySet;
   private final HostInfo hostInfo;
   private final IMonitorService monitorService;
-  private final AtomicLong connectionCheckIntervalMillis = new AtomicLong(DEFAULT_CONNECTION_CHECK_INTERVAL_MILLIS);
-  private final AtomicLong contextLastUsedTimestampNano = new AtomicLong();
-  private final AtomicBoolean stopped = new AtomicBoolean(true);
-  private final AtomicBoolean isConnectionCheckIntervalInitialized = new AtomicBoolean(false);
+  private volatile long contextLastUsedTimestampNano;
+  private boolean stopped = false;
```

Comment:
It's not necessary here. I cleaned it up.

Reply:
Not sure if you misunderstood, I was wondering if `stopped` needs to be volatile since it is read by the main thread and written by the monitor thread. `contextLastUsedTimeStampNano` is as well, so it may need to be volatile too, I'm not sure though

Reply:
Makes sense. Thank you.